catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -70,6 +70,20 @@
 
     private static final String DISTANCE_SORT_FIELD = "_distance_";
 
+    /**
+     * Because {@link SolrQuery#setRows(Integer)} will increment the given integer and use it as an
+     * array index, we cannot use {@link Integer#MAX_VALUE} by default and should pick a much lower,
+     * sensible ceiling for our row count. In practice, an un-paged query with a million or more results
+     * is not viable for the browser.
+     *
+     * Root problem is a lack of proper paging implementations across DDF; anywhere a
+     * {@link ddf.catalog.operation.impl.QueryImpl} is constructed with a pageSize < 1 is an issue that
+     * needs to be addressed.
+     *
+     * (See DDF-2872)
+     */
+    private static final Integer MAX_ROWS = 1000000;
```

Comment:
Might help to add separators in this number for readability: `MAX_ROWS = 1_000_000;`

Secondarily, any reason not to make this number a nice power of 2? (2^20) ~1mil.